### DISCUSSION TITLE
Spam Bot Protection

### DISCUSSION THREAD

**Discussion by vosiander** [open-webui/open-webui#6993] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**  
We see an influx of unwanted spam bot user registrations which try to exploit unsafe configurations
Currently, there is no adequate protection against spam bots during the registration process, leading to an influx of spam accounts. Additionally, there is no easy way for administrators to block specific email domains or manage email whitelists and blacklists effectively. This results in frustration when trying to maintain a secure and spam-free user base.

**Describe the solution you'd like**  
1. Implement spam bot protection mechanisms during registration, such as captchas or similar tools, to prevent automated account creation.  
2. Add admin functionality to block or restrict unwanted email domains with ease.  
3. Introduce options to maintain white and blacklists for email addresses, allowing for granular control over the registration process.  
4. Do not rely on first user creation equals admin. This should be provided via inital configuration/env-variables

**Describe alternatives you've considered**  
- Manually filtering and removing spam accounts, which is time-consuming and inefficient.  
- Using third-party email validation tools, which might not integrate seamlessly with the current system.  

**Additional context**  
Spam bot protection would significantly enhance the registration process's security and user experience. An admin-friendly interface for managing email restrictions will streamline the workflow for administrators, making it easier to combat spam. If needed, I can provide mockups or further details about the proposed features.

**Comment by stweil** [comment]
I had this problem, too, and decided to restrict registrations to users from a certain IP range.

Here is the configuration snippet for Apache 2.4 or newer:
```
  <Location "/api/v1/auths/signup">
    # Fix the following IP range to your needs.
    Require ip 123.456.0.0/16
    Require ip 127.0.0.1
    # Optionally add more conditions which allow registration, for example basic authentication.
  </Location>
```

This works as desired.

It could be improved by enhancing Open WebUI to produce a message which tells users that registration was forbidden for them. Maybe the registration button could also be removed conditionally.
